pkg/network/master/master.go

```diff
@@ -55,13 +55,16 @@ func Start(networkConfig osconfigapi.MasterNetworkConfig, networkClient networkc
 
 	var err error
 	var clusterNetworkEntries []networkapi.ClusterNetworkEntry
-	for _, cidr := range networkConfig.ClusterNetworks {
-		clusterNetworkEntries = append(clusterNetworkEntries, networkapi.ClusterNetworkEntry{CIDR: cidr.CIDR, HostSubnetLength: cidr.HostSubnetLength})
+	for _, entry := range networkConfig.ClusterNetworks {
+		clusterNetworkEntries = append(clusterNetworkEntries, networkapi.ClusterNetworkEntry{CIDR: entry.CIDR, HostSubnetLength: entry.HostSubnetLength})
 	}
 	master.networkInfo, err = common.ParseNetworkInfo(clusterNetworkEntries, networkConfig.ServiceNetworkCIDR)
 	if err != nil {
 		return err
 	}
+	if len(clusterNetworkEntries) == 0 {
+		panic("No ClusterNetworks set in networkConfig; should have been defaulted in if not configured")
```

Comment:
Do we want to panic here? returning an error will propagate to the SDN RunController() and will fail as expected.

Reply:
It can only happen if someone breaks the config-reading code. It's basically a "can't happen", I just put the check+panic in to make it slightly clearer what was going than if we just let them get an out-of-bounds exception when we dereference clusterNetworkEntries[0] below

Reply:
I wasn't suggesting allowing de-referencing clusterNetworkEntries[0] but to return a normal error instead of panic because it does the same job of stopping the sdn controller. I thought panic could be used when we are too deep in the code and just returning an error will not suffice (like stopping the controller).

Reply:
> I thought panic could be used when we are too deep in the code and just returning an error will not suffice (like stopping the controller).

Hm... I tend to think of it more as `error` is for runtime errors ("someone wrote bad values into the config file") and `panic` is for devel-time errors ("this code assumes that default values will have been filled in, but they weren't"). The Go spec is not especially clear about this, although "Effective Go" notes that `panic` can be "a way to indicate that something impossible has happened", which is what I was going for here.